stream.go

```diff
@@ -102,20 +102,16 @@ func (c *stream) Close() error {
 }
 
 func (c *stream) close(sendRST bool, readErr error, writeErr error) error {
-	didClose := false
 	c.mx.Lock()
 	if !c.closed {
 		c.closed = true
 		c.finalReadErr = readErr
 		c.finalWriteErr = writeErr
-		didClose = true
-	}
-	c.mx.Unlock()
-	if didClose {
 		c.rb.close()
 		c.sb.close(sendRST)
 		c.writeTimer.Stop()
 	}
+	c.mx.Unlock()
```

Comment:
I protected the whole close with the mutex so that we don't try to write to a closed channel [here](https://github.com/getlantern/lampshade/pull/7/files#diff-e2d91404d5609adc9273e4603f6ce67dR74).